addons/mail/static/src/js/thread.js

```diff
@@ -34,7 +34,8 @@ var Thread = Widget.extend({
             var message_id = $(event.currentTarget).data('message-id');
             this.trigger("mark_as_read", message_id);
         },
-        "click .o_thread_message_star": function (event) {
+        "click .o_thread_message_star:not(.disabled)": function (event) {
+            $(event.currentTarget).addClass('disabled');
```

Comment:
This is just a bandaid, but it does not fix the underlying issue.  When you encounter a problem, you need to understand the root cause before fixing it.  In this case, it it the chat_manager that calls the method 'set_message_starred', with the starred value of the message.  So, if you click quickly and the network is slow, you will send several requests to the server to change the star status to the same value.

So, in my opinion, the correct fix is to change the method 'set_message_starred' in mail_message into 'toggle_star_status', which does not take the star value in argument, but which does read it, then toggle it.  This method can take optionally the value in argument, but it should not be the way it is called by the chat manager.   This change means that if the user clicks quickly 3 times, then the server will simply toggle 3 times the star status, with the net result of just toggling it once, and the star counter should be correct.